utxo/src/utxo_entry.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Utxo;
+use serialization::{Decode, Encode};
+use std::fmt::Debug;
+
+/// Tells the state of the utxo
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+#[allow(clippy::large_enum_variant)]
+pub enum UtxoStatus {
+    Spent,
+    Entry(Utxo),
+}
+
+/// Just the Utxo with additional information.
+#[derive(Debug, Clone, Eq, PartialEq, Encode, Decode)]
+pub struct UtxoEntry {
+    status: UtxoStatus,
+    /// The utxo entry is dirty when this version is different from the parent.
+    is_dirty: bool,
+    /// The utxo entry is fresh when the parent does not have this utxo or
+    /// if it exists in parent but not in current cache.
+    is_fresh: bool,
+}
+
+impl UtxoEntry {
+    pub fn new(utxo: Option<Utxo>, is_fresh: bool, is_dirty: bool) -> UtxoEntry {
+        UtxoEntry {
+            status: match utxo {
+                Some(utxo) => UtxoStatus::Entry(utxo),
+                None => UtxoStatus::Spent,
+            },
+            is_dirty,
+            is_fresh,
+        }
+    }
+
+    pub fn is_dirty(&self) -> bool {
+        self.is_dirty
+    }
+
+    pub fn is_fresh(&self) -> bool {
+        self.is_fresh
+    }
+
+    pub fn is_spent(&self) -> bool {
+        self.status == UtxoStatus::Spent
+    }
+
+    pub fn utxo(&self) -> Option<&Utxo> {
```

Comment:
pay attention that this getter is now returning a ref instead of implicitly cloning every time

Reply:
It sounds like it should've been like that from the beginning. It's the right thing to do.